.github/workflows/infra_setup.yml

```diff
@@ -0,0 +1,140 @@
+name: Infrastructure setup
+
+run-name: ${{ github.actor }} is testing the pipeline
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [ "INTERN-77-" ]
+
```

Comment:
why workflow on dispatch here?

Reply:
Right now, it does not have any use case. But after merging the branch to main branch, it will allow to manually run the workflow.

Reply:
but it wont run automatically.

Reply:
Ok I have added another trigger as pull_request on main branch.